domain/containerimageresourcestore/service/service.go

```diff
@@ -69,49 +75,47 @@ func (s Service) Get(
 			},
 		},
 	}
-	data, err := json.Marshal(details)
-	if err != nil {
-		return nil, 0, errors.Errorf("marshelling container image metadata: %w", err)
-	}
-	infoReader := bytes.NewReader(data)
-	length := infoReader.Len()
-	return io.NopCloser(infoReader), int64(length), nil
+	return marshal(details)
```

Comment:
`marshal` is doing more than marshalling...

At the very least I want comments.